api/report.py

```diff
@@ -94,12 +96,21 @@ class ReportAPI(Resource):
     '''
     def put(self, reportid):
         '''
-        Get report by id
-        body: {
-            content: string
-            user: string
-        }
+        Update report content
+        body: Report
         '''
         report = Report.get(reportid)
+        user = User.get_from_request(request)
+        if report.group not in user.groups:
+            raise Forbidden()
+        if 'authors' in api.payload:
+            api.payload.pop('authors')
         report.update(api.payload)
+        if 'content' in api.payload:
+            previous_authors = []
+            if report.authors:
+                previous_authors = [author for author in report.authors if author.id != user.id]
+            new_authors = [user] + previous_authors
```

Comment:
What is the point of first filtering out the user from the authors and then adding it back?

Reply:
- To make the most recent author who edit the report be the fist in the list.
- To prevent duplicate users if they edit the report more than one time.